tapdb/assets_store.go

```diff
@@ -2745,6 +2756,47 @@ func (a *AssetStore) QueryProofTransferLog(ctx context.Context,
 	return timestamps, err
 }
 
+// ConfirmProofDelivery marks a transfer output proof as successfully
+// delivered to counterparty.
+func (a *AssetStore) ConfirmProofDelivery(ctx context.Context,
+	anchorOutpoint wire.OutPoint, outputPosition uint64) error {
+
+	// Serialize the anchor outpoint to bytes.
+	anchorOutpointBytes, err := encodeOutpoint(anchorOutpoint)
+	if err != nil {
+		return fmt.Errorf("unable to encode anchor outpoint: %w", err)
+	}
+
+	// Ensure that the position value can be stored in a 32-bit integer.
+	// Type cast if possible, otherwise return an error.
+	if outputPosition > math.MaxInt32 {
+		return fmt.Errorf("position value is too large for db: %d ",
+			outputPosition,
+		)
```

Comment:
nit: closing parenthesis on previous line for function call with formatting directive (also, there's an extra space at the end of the string).